### PR TITLE
Update Project Architecture Chapter

### PR DESCRIPTION
**Pull request by kristinakoneva**
Task: [#636 Update MVVM chapter](https://app.productive.io/1-infinum/projects/2590/tasks/task/1749329?filter=MjQ1OTU5)

Summary of the things I added in the MVVM chapter:
- I added some info about Flows, SharedFlows, StateFlows since they are a good alternative to using LiveData. I added a small section comparing these types with LiveData.
- In the “Implementation” section, I added a subsection “Implementation using StateFlow and SharedFlow” with some examples how LiveData can be substituted.
- I wrote a short section where I compared Coroutines with RxJava.

Additional notes:
- I deleted some images from the Project architecture folder since they were already present in the img folder.
- I guess the “What is solved with MVVM?” section of the MVVM chapter should also be updated soon since, at the moment, it is only referring to the implementation using ViewModels with LiveData.
- At the end of the MVVM chapter, there is a link to the Android-Architecture-Example Github repo. However, since this repo is private, when the link is clicked from the public handbook, an Error 404 is shown.

### CODE REVIEW

**Comment by ffloreani** [comment]
Thanks for the effort, Kristina! ❤️

I've added all the LEs here, but it's enough that only one of you guys review this 😄

File: Project architecture/MVVM.md
Comment (KCeh) [inline Project architecture/MVVM.md:85]: Great chapter!
But should this really be under MVVM? Common Android?

File: Project architecture/MVVM.md
Comment (KCeh) [inline Project architecture/MVVM.md:58]: Should LiveData, Flow etc be in a separate file?
Common Android? And just reference their use here.
This would go into a more extensive discussion about the restructuring handbook. All comments are welcomed.

File: Project architecture/MVVM.md
Comment (KCeh) [inline Project architecture/MVVM.md:191]: Again, this should go into a more extensive discussion about the restructuring handbook. I think we should add chapter about the common testing setup for architecture (base test classes, how to test livedata, flows etc)
I remember recently there was some question about it in #team-android

File: Project architecture/MVVM.md
Comment (KCeh) [reply Project architecture/MVVM.md:93]: Also, maybe a bit different phrasing would be better?
```suggestion
Using coroutines and Kotlin for Android development goes hand in hand. If you are using RxJava our suggestion would be to migrate to coroutines. To help you with migration you can covert RxJava code to coroutines using libraries like [kotlinx-coroutines-rx3](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-rx3/)
```

**Review by KCeh — COMMENTED** [review]
Great job Kristina 👏 🥳 
I left a few comments but most are open for discussion and related to handbook restructuring. You can also give your opinion on them 😄

File: Project architecture/MVVM.md
Comment (KCeh) [inline Project architecture/MVVM.md:219]: @ffloreani Kristina mentioned this in the PR description. Here is a link to the private Infinum repo in the public Handbook. Do we keep it?

File: Project architecture/MVVM.md
Comment (kristinakoneva) [reply Project architecture/MVVM.md:85]: I agree, it is a bit out of place in the MVVM chapter, maybe we could move it to some other chapter as you suggested.

File: Project architecture/MVVM.md
Comment (ffloreani) [reply Project architecture/MVVM.md:219]: Sorry for taking so long to answer 😅 yeah, we should not have any links from the public to the private handbook. It breaks our SEO and page rank on Google. We can just remove the link and we should be good 😄

Comment (kristinakoneva) [reply Project architecture/MVVM.md]: I removed it. 👍 I also updated some links in the MVP chapter which were linking existing chapters, but I guess some restructuring of the handbook was done in the meantime, so the links were giving me “This chapter does not exist.” error even when I was logged in with Infinum ID. One more thing 😅 , in the Mobile app architecture chapter, in the Documentation section, there are some links to keynote presentations on Infinum’s Google drive which are not available to the public (only the link to the PDF template is accessible), so I guess parts of that section should be removed as well. We could consider relocating this section somewhere in the private repo chapters because it seems useful.

Comment (KCeh) [reply Project architecture/MVVM.md]: That sounds like a good idea to me

Comment (ffloreani) [reply Project architecture/MVVM.md]: I'd really love to keep the _Mobile app architecture_ page public, it's good content that can attract other devs to Infinum 👿 

A fun fact is that links that are hidden behind Google's own login are not counted as broken links 😄 so no worries about things stored on our Google Drive

Comment (kristinakoneva) [reply Project architecture/MVVM.md]: I meant moving parts of the Documentation section to the private repo (the steps with instructions “How to” looked very Infinum specific to me), definitely didn’t think about moving the whole chapter. 😅 However, yeah, since they don’t count as broken links, no biggie, they might as well stay. 😄 Make the devs come to Infinum if they want to see where the links go? 😉

File: Project architecture/Mobile app architecture.md
Comment (stjepanbanek) [inline Project architecture/Mobile app architecture.md:153]: Not a big deal, but the mapper in this constructor has `...Mapper` suffix in the name while interactors don't. For consistency it would be nice to stick to one way of naming thing, whatever way that is.

File: Project architecture/Mobile app architecture.md
Comment (stjepanbanek) [inline Project architecture/Mobile app architecture.md:171]: A few minor points here.
1. I think this section would be more readable and easier to keep reader's attention if it had some line breaks, as it is a longer chunk of text. 
2. We mention how repository only has interactors and mappers here. This might not always be true, but the main purpose of repository should always stay the same. For example: In my project I use classes called `Converters` to map stuff from domain layer to the source layer. I would go more abstract than to name classes here and instead describe the purpose of a repository -> In best case scenario it aggregates all classes enabling CRUD operations for one domain model. In some cases maybe 2. This means that it may contain interactors, mappers, converters etc. But that set might get bigger. And the class naming might change especially with the new mappers as extensions. The main point though, should stay the same, so I would personally rather focus on explaining the point of the repository using classes here as examples, but pointing out that those are not concrete.

What do you think?

**Review by stjepanbanek — CHANGES_REQUESTED** [review]
Great job @kristinakoneva!

I've left a few minor suggestions, but feel free to express your opinion on them even if you don't agree with me always. Once we resolve them, I will gladly approve even if we stick to the current texts.

File: Project architecture/Mobile app architecture.md
Comment (kristinakoneva) [reply Project architecture/Mobile app architecture.md]: 1. Done ✅ 
2. I added a sentence or two clearing this up. I used what you mentioned about the converters as an example of how the repository implementation could differ. 😁 

In this PR, I didn’t really do any major changes in the Mobile app architecture chapter since it was already well-written and still very applicable to today’s architecture. 🚀

**Review by stjepanbanek — APPROVED** [review]
Great work! 🚀

**Comment by KCeh** [comment]
I guess we are good to merge.
The potential restructuring of MVVM chapter will be addressed in future PRs.
Thanks for great contribution @kristinakoneva and please open PR to update main with staging 😄

**Pull request merged**
